contrib/gcppubsub/pkg/adapter/adapter.go

```diff
@@ -98,23 +111,36 @@ func (a *Adapter) receiveMessage(ctx context.Context, m PubSubMessage) {
 	}
 }
 func (a *Adapter) postMessage(ctx context.Context, logger *zap.SugaredLogger, m PubSubMessage) error {
+	// Create the CloudEvent.
+	event := cloudevents.NewEvent(cloudevents.VersionV02)
+	event.SetID(m.ID())
+	event.SetTime(m.PublishTime())
+	event.SetDataContentType(*cloudevents.StringOfApplicationJSON())
+	event.SetSource(a.source)
+	event.SetData(m.Message())
+
 	// TODO: this will break when the upstream sender updates cloudevents versions.
 	// The correct thing to do would be to convert the message to a cloudevent if it is one.
 	et := sourcesv1alpha1.GcpPubSubSourceEventType
 	if override, ok := m.Message().Attributes[eventTypeOverrideKey]; ok {
```

Comment:
IMO we should remove this asap. If a particular source wants to do something like this (e.g., the gcs source), it should actually do it in the transformer.

Reply:
I agree.